distributed/comm/addressing.py

```diff
@@ -9,16 +9,21 @@
 DEFAULT_SCHEME = config.get('default-scheme', 'tcp')
 
 
-def parse_address(addr):
+def parse_address(addr, strict=False):
     """
     Split address into its scheme and scheme-dependent location string.
 
     >>> parse_address('tcp://127.0.0.1')
     ('tcp', '127.0.0.1')
+
+    If strict is set to true the address must have a scheme.
     """
     if not isinstance(addr, six.string_types):
         raise TypeError("expected str, got %r" % addr.__class__.__name__)
     scheme, sep, loc = addr.rpartition('://')
+    if strict and not sep:
+        raise ValueError("Invalid url scheme. Must be of the form <protocol>://<host>:<port>. "
+                        "Got %s" % addr)
```

Comment:
Regarding testing, please take a look at the tests in `distributed/cli/tests/test_dask_worker.py`.  You may want to use `client.scheduler_info()` to get information about the scheduler and workers.  You can also check for various attributes using [client.run](http://distributed.readthedocs.io/en/latest/api.html#distributed.Client.run)

> I believe currently, this is explicit. You need to specify the ports. I will post an update after checking.

So there aren't situations where workers would want to identify themselves by different hostnames but wouldn't know their ports ahead of time?  

> I understand this. But the way the cli/dask_worker.py is written it expects a port? 

This was true historically though I think that @pitrou has been trying to move away from this and keep addresses general.  For example we've been asked to look into supporting communication through other protocols like MPI.  This hasn't happened yet of course, so it's not something that you should feel obliged to optimize for, but it would be best to make sure that we don't exclude in the future.